src/upgrade_step_check_init_inst.py

```diff
@@ -0,0 +1,69 @@
+from src.file_access import FileAccess
+from src.local_logger import LocalLogger
+from src.upgrade_step import UpgradeStep
+from src.common_upgrades.utils.constants import CONFIG_ROOT
+import os
+# from io import open
+
+
+class UpgradeStepCheckInitInst(UpgradeStep):
+    """
+    An upgrade step to check if the instrument uses the old style of loading in pre and post cmd.
+    This old style is via API.__localmod in init_<inst>.py in the Instrument/Settings/config/NDX<inst>/Python folder. 
+    """
+
+    def search_files(self, files, root, file_access):
+        """
+        Search files from a root folder for pre and post cmd methods.
+
+        Args:
+            files (List[str]): The names of the files in the root directory.
+            root (str): The root directory of the files.
+            file_access (FileAccess): file access
+        
+        Returns: 0 if pre and post cmd methods in old style are not present; error message if they are.
+        """
+        for file_name in files:
+            if file_name.startswith("init_"):
+                search_file = open(os.path.join(root, file_name))
+                search_file_contents = search_file.read()
+                if "precmd" in search_file_contents or "postcmd" in search_file_contents:
+                    return "Pre or post cmd methods found in {} convert to new style of inserting these methods".format(search_file.name)
```

Comment:
Can you be a bit more explicit on how to do this or link to the wiki page that shows you how.

Reply:
Link added which describes the pre and post command hook: https://github.com/ISISComputingGroup/ibex_user_manual/wiki/genie_python-Commands